src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+        break;
+      case NB_BATHROOMS:
+        prof.setNbBathroomsSubjectiveValueWeight(value);
+        break;
+      case NB_BEDROOMS:
+        prof.setNbBedroomsSubjectiveValueWeight(value);
+        break;
+      case NB_SLEEPING:
+        prof.setNbSleepingSubjectiveValueWeight(value);
+        break;
+      case NB_MIN_NIGHT:
+        prof.setNbMinNightSubjectiveValueWeight(value);
+        break;
+      case PRICE_PER_NIGHT:
+        prof.setPricePerNightSubjectiveValueWeight(value);
+        break;
+      default:
+        throw new IllegalArgumentException();
+    }
+
+    return prof;
+  }
+
+  /**
+   * check if the given range is valid for this context
+   *
+   * @param value the Range to check
+   * @throws IllegalArgumentException when doubles contained are < 0 or if there are only 1 value in
+   *     the range
+   */
+  private void checkRangeValidity(Range<Double> value) {
+    checkNotNull(value, "The rage cannot be null");
+    checkArgument(value.hasLowerBound() && value.hasUpperBound(), "The given range is not valid");
+    checkArgument(
+        value.lowerEndpoint() >= 0 && value.upperEndpoint() > 0,
+        "The weight of the tele cannot be negative");
+  }
+
+  /**
+   * Set the weight of the floor area subjective value corresponding to the importance of the floor
+   * area criteria.
+   *
+   * @param value >= 0
+   */
+  private void setFloorAreaSubjectiveValueWeight(Range<Double> value) {
+    checkRangeValidity(value);
+    this.floorAreaWeightRange = value;
+    this.linearAvf = this.linearAvf.setWeight(Criterion.FLOOR_AREA, getMiddleOfRange(value));
+    LOGGER.debug("The floor area weight has been set to {}", value);
+  }
+
+  /**
+   * Set the weight of the number of bedrooms subjective value corresponding to the importance of
+   * the number of bedrooms criteria.
+   *
+   * @param value >= 0
+   */
+  private void setNbBedroomsSubjectiveValueWeight(Range<Double> value) {
+    checkRangeValidity(value);
+    this.nbBedroomsWeightRange = value;
+    this.linearAvf = this.linearAvf.setWeight(Criterion.NB_BEDROOMS, getMiddleOfRange(value));
+    LOGGER.debug("The number of bedrooms weight has been set to {}", value);
+  }
+
+  /**
+   * set the weight of the number of sleeping subjective value corresponding to the importance of
+   * the number of sleeping criteria.
+   *
+   * @param value >= 0
+   */
+  private void setNbSleepingSubjectiveValueWeight(Range<Double> value) {
+    checkRangeValidity(value);
+    this.nbSleepingWeightRange = value;
+    this.linearAvf = this.linearAvf.setWeight(Criterion.NB_SLEEPING, getMiddleOfRange(value));
+    LOGGER.debug("The number of sleeping weight has been set to {}", value);
+  }
+
+  /**
+   * Set the weight of the number of bathrooms subjective value corresponding to the importance of
+   * the number of bathrooms criteria.
+   *
+   * @param value >= 0
+   */
+  private void setNbBathroomsSubjectiveValueWeight(Range<Double> value) {
```

Comment:
Cette duplication de code peu élégante me semble pouvoir être évitée grâce à votre type Criterion.

Reply:
Fixed in commit 5e49a48